services/movie_session.py

```diff
@@ -0,0 +1,57 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int,
+) -> MovieSession:
+    new_movie_session = MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=CinemaHall.objects.get(id=cinema_hall_id),
```

Comment:
no need to pass objects here - you can pass ids directly

Reply:
When I  did it i get an error:  ValueError: Cannot assign "2": "MovieSession.cinema_hall" must be a "CinemaHall" instance.

Reply:
It's because you have explicit field cinema_hall and movie(that you added in the models). But under the hood in database it's saved as cinema_hall_id and movie_id because database doesn't know what is Django and related models. So you can use these "hidden" fields and set id instead of making redundant queries
@ihorutkin